src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -146,16 +147,32 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
    */
   ColumnID get_column_id(const AbstractExpression& expression) const;
 
+  /**
+   * @return True, if the given set of expressions is a subset of the node's output expressions. False otherwise.
+   */
+  bool has_output_expressions(const ExpressionUnorderedSet& expressions) const;
+
   /**
    * @return whether the output column at @param column_id is nullable
    */
   virtual bool is_column_nullable(const ColumnID column_id) const;
 
+  /**
+   * @return Unique constraints valid for the current LQP. See lqp_unique_constraint.hpp for more documentation.
+   */
+  virtual std::shared_ptr<LQPUniqueConstraints> unique_constraints() const = 0;
+
+  /**
+   * @return True, if there is unique constraint matching the given subset of output expressions.
+   *         (i.e., the rows are guaranteed to be unique).
```

Comment:
Typo: If there is **a**...

Also: Preferred to calling `contains_matching_unique_constraint` directly, as it performs additional sanity checks.